pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,223 @@
+        volume = np.tile(volume, len(rotations))
+        C = np.einsum('n,ni,nj->nij', 0.5*volume, strain, strain)
+        C = np.triu(C).reshape(-1, 36)
+        C = C[:,np.sum(C, axis=0)!=0]
+        C = np.append(np.ones(len(C)).reshape(-1, 1), C, axis=-1)
+        reg = LinearRegression().fit(C, energy)
+        coeff = np.triu(np.ones((6,6))).flatten()
+        coeff[coeff!=0] *= reg.coef_[1:]*eV_div_A3_to_GPa
+        coeff = coeff.reshape(6,6)
+        coeff = 0.5*(coeff+coeff.T)
+    else:
+        raise ValueError('Problem with fitting data')
+    return coeff
+
+def calc_elastic_constants(elastic_tensor):
+    """
+    Calculate elastic constants from the elastic tensor.
+
+    For anistropic material (i.e. zener_ratio!=1), the values may or may not make
+    sense -> don't trust the results straightforwardly
+
+    Args:
+        elastic_tensor (numpy.ndarray): 6x6 tensor
+    """
+    output = {}
+    output['elastic_tensor'] = elastic_tensor
+    output['lame_coefficient'] = np.mean(elastic_tensor[:3, :3].diagonal())
+    output['shear_modulus'] = np.mean(elastic_tensor[3:, 3:].diagonal())
+    output['bulk_modulus'] = np.mean(elastic_tensor[:3,:3])
+    output['poissons_ratio'] = 0.5/(1+output['shear_modulus']/output['lame_coefficient'])
+    output['youngs_modulus'] = 1/np.mean(np.linalg.inv(elastic_tensor[:3,:3]).diagonal())
+    output['poissons_ratio'] = -output['youngs_modulus']*np.sum(np.linalg.inv(elastic_tensor[:3,:3]))/6+0.5
+    output['zener_ratio'] = 12*np.mean(elastic_tensor[3:,3:].diagonal())/(3*np.trace(elastic_tensor[:3,:3])-np.sum(elastic_tensor[:3,:3]))
+    return output
+
+class ElasticJobGenerator(JobGenerator):
+    @property
+    def parameter_list(self):
+        """
+
+        Returns:
+            (list)
+        """
+        parameter_lst = []
+        for ii, epsilon in enumerate(self._job.input['strain_matrices']):
+            basis = self._job.ref_job.structure.copy()
+            basis.apply_strain(np.array(epsilon))
+            parameter_lst.append([ii, basis])
+        return parameter_lst
+
+    def job_name(self, parameter):
+        return "{}_{}".format(self._job.job_name, parameter[0]).replace(".", "_")
+
+    @staticmethod
+    def modify_job(job, parameter):
+        job.structure = parameter[1]
+        return job
+
+# ToDo: not all abstract methods implemented
+class ElasticTensor(AtomisticParallelMaster):
+    """
+    Class to calculate the elastic tensor and isotropic elastic constants
+
+    Example:
+
+    >>> job = pr.create_job('SomeAtomisticJob', 'atomistic')
+    >>> job.structure = pr.create_structure('Fe', 'bcc', 2.83)
+    >>> elastic = job.create_job('ElasticTensor', 'elastic')
+    >>> elastic.run()
+
+    This class is still under construction and there's no guarantee that it does
+    what it looks like it does.
+    """
+    def __init__(self, project, job_name="elastic"):
+        """
+
+        Args:
+            project:
```

Comment:
Boooo missing docstrings! ;)

Reply:
uuuuuuuuh but do we need a docstring in `__init__`?

Reply:
I guess not, but then let's clean up the stub. Same for the job generator. If there's no change relative to the parent I don't see a need for a docstring, but what's there now is just clutter.